gitops/snapshot.go

```diff
@@ -168,6 +174,9 @@ func MarkSnapshotAsFailed(adapterClient client.Client, ctx context.Context, snap
 
 // SetSnapshotIntegrationStatusAsInvalid sets the AppStudio integration status condition for the Snapshot to invalid.
 func SetSnapshotIntegrationStatusAsInvalid(snapshot *applicationapiv1alpha1.Snapshot, message string) {
+	if !IsSnapshotValid(snapshot) {
```

Comment:
I still think we could make this check in the adapter code before calling `IsSnapshotValid()`.